py/objint.c

```diff
@@ -300,6 +300,49 @@ char *mp_obj_int_formatted(char **buf, size_t *buf_size, size_t *fmt_size, mp_co
     return b;
 }
 
+void mp_obj_int_buffer_overflow_check(mp_obj_t self_in, size_t nbytes, bool is_signed)
+{
+    if (is_signed) {
+        // edge = 1 << (nbytes * 8 - 1)
+        mp_obj_t edge = mp_binary_op(MP_BINARY_OP_INPLACE_LSHIFT,
+                                     mp_obj_new_int(1),
+                                     mp_obj_new_int(nbytes * 8 - 1));
+
+        // if self >= edge, we don't fit
+        if (mp_binary_op(MP_BINARY_OP_MORE_EQUAL, self_in, edge) == mp_const_true) {
+            goto raise;
```

Comment:
No gotos please. I think you can invert the checks and return if the value is ok instead.

Reply:
Are early returns OK?  I can refactor this into 2 functions, one checking if the value is valid (with early returns if it's not) and one raising for invalid values.

Also: do I need to do any memory management here, or will the GC take care of cleaning up any big ints that I'm creating here?  I tried calling this in a loop and it didn't seem to leak (or, rather, the memory climbed until I called `gc.collect` and then dropped back down) - but I don't understand _how_ it worked, so I might still be missing something.

Reply:
All python object are cleaned up automatically, unless specifically prevented from doing that.

Reply:
@godlygeek Yup, early returns are fine. Collect will clean the large ints up ok as @deshipu points out.